Работа по спринту 14

UtinsStory opened the pull request.

UtinsStory committed:
Сверстаны экраны трекеров и статистики. Необходимо доделать внешний вид datePicker согласно ТЗ

UtinsStory committed:
Созданы модели для трекера

UtinsStory committed:
Главный экран сверстан кодом. Реализован функционал кнопки добавления трекеров

UtinsStory committed:
Продолжена работа по реализации функционала создания привычки

UtinsStory committed:
Реализован функционал создания привычки

UtinsStory committed:
Реализован функционал создания нерегулярного события

UtinsStory committed:
Фикс бага сборки проекта

UtinsStory committed:
фикс показа заглушек при пустом списке трекеров

UtinsStory committed:
Переделана логика создания трекеров, показа заглушки, формирования дней недели. Задание спринта 14 готов к сдаче

DanilaG commented on TrackerApp/TrackerApp/Tracker/TrackerViewController.swift:
```diff
@@ -0,0 +1,480 @@
+            return
+        }
+
+        if let index = completedTrackers.firstIndex(where: { isSameTrackerRecord(trackerRecord: $0, id: id) }) {
```
Можно упростить до:
`completedTrackers.removeAll(where: { isSameTrackerRecord(trackerRecord: $0, id: id) })`

DanilaG commented on TrackerApp/TrackerApp/Tracker/TrackerViewController.swift:
```diff
@@ -0,0 +1,480 @@
+        dateLabel.text = dateString
+    }
+
+    func reloadFilteredCategories(text: String?, date: Date) {
```
Обычно, не приватные функции идут перед приватными. Давай поднимим функцию

UtinsStory committed:
Фикс ошибок после ревью:
-Добавлена логика отображения UICollectionView для различных версий iOS.
-Переменная currentDate хранит значение согласно ТЗ.
-Корректно создаются и отображаются привычки и неругялрные события в различные дни.
*Добавлена возможность скрытия виртуальной клавиатуры по тапу

UtinsStory committed:
Экстренный фикс показа хэдера в коллекции

DanilaG commented on TrackerApp/TrackerApp/Helpers/UIViewContoller+Extensions.swift:
```diff
@@ -0,0 +1,19 @@
+import UIKit
+
+extension UIViewController {
+    func hideKeyboardWhenTappedAround() {
```
🔥
